atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/RelationalDbTimeLockFactory.java

```diff
@@ -59,15 +63,23 @@ public KeyValueService createRawKeyValueService(
     @Override
     public ManagedTimestampService createManagedTimestampService(
             KeyValueService rawKvs,
-            Optional<DbTimestampCreationSetting> dbTimestampCreationSetting,
+            DbTimestampCreationSetting dbTimestampCreationSetting,
             boolean initializeAsync) {
-        return delegate.createManagedTimestampService(rawKvs, dbTimestampCreationSetting, initializeAsync);
+        return DbTimestampCreationSettings.caseOf(dbTimestampCreationSetting)
+                .multipleSeries((un, used) ->
+                        delegate.createManagedTimestampService(
+                                rawKvs, Optional.of(dbTimestampCreationSetting), initializeAsync))
+                .singleSeries(unused -> {
+                    throw new SafeIllegalStateException("DB TimeLock cannot be used with"
```

Comment:
You're right that we currently call into this method only with multiSeries. DbKVS _does_ accept singleSeries too, though (it has to for large internal product), unless I'm missing something, and I'd want to be resilient to anyone changing this in the future without knowing this.

... that said, based on this I think we can simplify the design here substantially - because we split `DbAtlasDbFactory` and `RelationalDbTimeLockFactory`, we don't actually need the DbTimestampCreationSetting at all I think. Probably not for this PR, and it does not need to be finished this week, but we should get that in before we call this workstream complete.